tests/test_models.py

```diff
@@ -1,37 +1,239 @@
 """
-Test cases for YourResourceModel Model
+Test cases for Inventory Model
 
 """
 import os
 import logging
 import unittest
-from service.models import YourResourceModel, DataValidationError, db
+from werkzeug.exceptions import NotFound
+from tests.factories import InventoryFactory
+from service.models import Inventory, Condition, DataValidationError, db
+from service import app
 
+DATABASE_URI = os.getenv(
+    "DATABASE_URI", "postgresql://postgres:postgres@localhost:5432/testdb"
+)
 
 ######################################################################
-#  YourResourceModel   M O D E L   T E S T   C A S E S
+#  Inventory   M O D E L   T E S T   C A S E S
 ######################################################################
-class TestYourResourceModel(unittest.TestCase):
-    """ Test Cases for YourResourceModel Model """
+class TestInventory(unittest.TestCase):
+    """ Test Cases for Inventory Model """
 
     @classmethod
     def setUpClass(cls):
-        """ This runs once before the entire test suite """
+        """This runs once before the entire test suite"""
+        app.config["TESTING"] = True
+        app.config["DEBUG"] = False
+        app.config["SQLALCHEMY_DATABASE_URI"] = DATABASE_URI
+        app.logger.setLevel(logging.CRITICAL)
+        Inventory.init_db(app)
 
     @classmethod
     def tearDownClass(cls):
-        """ This runs once after the entire test suite """
+        """This runs once after the entire test suite"""
+        db.session.close()
 
     def setUp(self):
-        """ This runs before each test """
+        """This runs before each test"""
+        db.session.query(Inventory).delete()  # clean up the last tests
+        db.session.commit()
 
     def tearDown(self):
-        """ This runs after each test """
+        """This runs after each test"""
+        db.session.remove()
 
     ######################################################################
     #  T E S T   C A S E S
     ######################################################################
 
-    def test_example_replace_this(self):
-        """ It should always be true """
-        self.assertTrue(True)
+    def test_create_a_inventory(self):
+        """It should Create a Inventory and assert that it exists"""
+        inventory = Inventory(product_id=1, condition= Condition.NEW, quantity=1, restock_level=3)
+        self.assertEqual(str(inventory), "<Inventory product_id=[1] condition=[Condition.NEW]>")
+        self.assertTrue(inventory is not None)
+        self.assertTrue(inventory.product_id is not None)
+        self.assertEqual(inventory.condition, Condition.NEW)
+        self.assertEqual(inventory.quantity, 1)
+        self.assertEqual(inventory.restock_level, 3)
+
+        inventory = Inventory(product_id=2, condition=Condition.OPEN_BOX, restock_level=5)
+        self.assertEqual(inventory.quantity, None)
```

Comment:
Oh yes, I actually wanted to discuss this on call. Since I did not specify any value for quantity, 

- should we assign the default value of `1` at db level **OR**
- should we handle that at API level, say by throwing an error that user must pass some value for quantity and reject the API request?

Reply:
So I have added the default value for `quantity `and `restock_level` as `1` at schema level. So when you create an object without specifying the value, they will be `None`. But when the record gets added to DB, then the default value of `1` will be used.